compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -987,6 +974,20 @@ DistributionLayout *EnforceLayout::getLatticeElement(Value val) {
 ///        VectorLayoutAnalysis
 /// ==========================================================================
 
+LogicalResult VectorLayoutAnalysis::setAnchor(Value val, Attribute layout) {
+  VectorLayoutInterface layoutInterface =
+      dyn_cast<VectorLayoutInterface>(layout);
+  assert(layoutInterface &&
+         "expected layout to implement VectorLayoutInterface");
+  auto typedVal = dyn_cast<TypedValue<VectorType>>(val);
+  assert(typedVal && "expected value to be a vector type");
```

Comment:
I think it would be better to take the `VectorLayoutInterface` directly as an argument instead of casting. IIUC most places where we set anchors, we're directly constructing a layout attr. Maybe for the typed value too? I could go either way on that one though.

Reply:
I changed VectorLayoutInterface to be the argument type. I think I'll do TypedValue for a separate patch since it will involve changing things at a lot of places.

Reply:
Sounds good, but I could go either way. If it requires changing in a lot of places where we basically are just moving the cast out of the `setAnchor` function, probably not worth doing. I'll approve whatever you pick though :)

Reply:
Sounds good :)